Marlin/src/module/planner.cpp

```diff
@@ -801,10 +801,8 @@ void Planner::calculate_trapezoid_for_block(block_t * const block, const_float_t
   // should protect against it. But removing results in less smooth motion for switching direction
   // moves. This is because the current discrete stepping math diverges from physical motion under
   // constant acceleration when acceleration_steps_per_s2 is large compared to initial/final_rate.
-  NOLESS(initial_rate, uint32_t(MINIMAL_STEP_RATE));
-  NOLESS(final_rate, uint32_t(MINIMAL_STEP_RATE));
-  NOMORE(initial_rate, block->nominal_rate);
-  NOMORE(final_rate, block->nominal_rate);
+  LIMIT(initial_rate, uint32_t(MINIMAL_STEP_RATE), block->nominal_rate);
```

Comment:
LIMIT assumes the given bounds are correctly ordered whereas MINIMAL_STEP_RATE can be larger than block->nominal_rate.

Reply:
Thanks for pointing that out. It's a case that can be very easy to overlook in this kind of "automatic" refactor.